modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -702,14 +748,31 @@ public boolean setTabSize(int spaces) {
      *                                                                         *
      **************************************************************************/
 
-    private int getLineIndex(float y) {
+    private int getLineIndex(float y, String text) {
         int index = 0;
         float bottom = 0;
+        boolean isTextPresent = text == null ? true : false;
+        boolean isPresent = true;
         int lineCount = getLineCount();
         while (index < lineCount) {
+            if (text != null) {
+                for (TextRun r: lines[index].runs) {
+                    if (r.getTextSpan() == null) {
+                        isTextPresent = true;
+                        break;
+                    } else if (r.getTextSpan().getText().equals(text)) {
+                        if (isPresent) {
+                            bottom = 0;
+                            isPresent = false;
+                        }
+                        isTextPresent = true;
+                        break;
+                    }
+                }
+            }
             bottom += lines[index].getBounds().getHeight() + spacing;
             if (index + 1 == lineCount) bottom -= lines[index].getLeading();
-            if (bottom > y) break;
+            if (bottom > y && isTextPresent) break;
```

Comment:
I would recommend adding { }'s to ifs here and on line 774.

Reply:
Added {} as suggested